reference-delegate-explainer.md

```diff
@@ -0,0 +1,502 @@
+    </div>
+  </template>
+</fancy-listbox>
+```
+
+Declaring the mappings using the JavaScript API would look like:
+```js
+this.shadowRoot_.referenceDelegateMap['aria-controls'] = 'real-listbox';
+this.shadowRoot_.referenceDelegateMap['aria-activedescendant'] = 'option-2';
+```
+
+> Note: the syntax of `shadowrootreferencedelegatemap` is based on the [`exportparts`](https://drafts.csswg.org/css-shadow-parts/#exportparts-attr) attribute that contains a comma-separated map of part names.
+
+#### Combining `referenceDelegate` and `referenceDelegateMap`
+
+In the case where both attributes are specified, `referenceDelegateMap` takes priority for individual attributes, and `referenceDelegate` acts as the fallback for attributes that are not specified.
+
+In the example below, `"real-listbox"` is the delegate for all attributes _except_ `aria-activedescendant`, which is delegated to `"option-2"`.
+
+```html
+<input aria-controls="fancy-listbox"
+       aria-activedescendant="fancy-listbox" />
+<fancy-listbox id="fancy-listbox">
+  <template shadowrootmode="open"
+            shadowrootreferencedelegate="real-listbox"
+            shadowrootreferencedelegatemap="aria-activedescendant: option-2">
+    <div id="real-listbox" role="listbox">
+      <div id="option-1" role="option">Option 1</div>
+      <div id="option-2" role="option">Option 2</div>
+    </div>
+  </template>
+</fancy-listbox>
+```
+
+#### Delegating to multiple elements
+
+Some attributes such as `aria-labelledby`, `aria-describedby`, etc. support multiple targets. Mappings for those attributes support a space-separated list of IDs.
+
+This example shows a `<description-with-tooltip>` component that contains a "More Info" button to show the tooltip but is not intended to be included in the description text. It delegates `aria-describedby: message tooltip` to forward to only the content that should be included in the description text.
+
+```html
+<input aria-describedby="description-with-tooltip" />
+<!--
+  The resulting description text is: 
+  "Inline description text. Tooltip with more information."
+-->
+<description-with-tooltip id="description-with-tooltip">
+  <template shadowrootmode="closed"
+            shadowrootreferencedelegatemap="aria-describedby: message tooltip">
+    <div>
+      <span id="message">
+        Inline description text.
+      </span>
+      <button onmouseover="showTooltip()" onmouseout="hideTooltip()">
+        More Info
+      </button>
+      <div id="tooltip" role="tooltip" style="display: none">
+        Tooltip with more information.
+      </div>
+    </div>
+  </template>
+</description-with-tooltip>
+```
+
+### Supported attributes
+
+This feature is intended to work with **all** attributes that refer to another element by ID string. These are:
+
+* `aria-activedescendant`
+* `aria-controls`
+* `aria-describedby`
+* `aria-details`
+* `aria-errormessage`
+* `aria-flowto`
+* `aria-labelledby`
+* `aria-owns`
```

Comment:
Should include the `popover` and “invoker” attributes here?

Reply:
Good call -- yes it should include them. I've also added a few other attributes to the list that I missed earlier.